File: web/server/mongodb/models/LogEvent.ts
import { LogEvent } from '@/utils/types';
import mongoose, { Schema } from 'mongoose';

const LogEventSchema = new Schema<LogEvent>(
    {
        logType: {
            type: String,
            required: true,
        },
        workspaceId: {
            type: Schema.ObjectId,
            required: true,
        },
        blockchainAddress: {
            type: String,
            required: true,
        },
        xDataPoint: {
            type: Number,
            required: true,
        },
        yDataPoint: {
            type: Number,
            required: true,
        },
    },
    {
        timestamps: {
            createdAt: true,
            updatedAt: false,
        },
    }
);

export const LogEventModel =
    (mongoose.models.LogEvent as mongoose.Model<LogEvent>) ??
    mongoose.model<LogEvent>('LogEvent', LogEventSchema);

File: core/logs/log_event.go
package scatterlogs

import (
	"context"
	"log"
	"time"

	"github.com/gtfintechlab/scatter-protocol/core/utils"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

func CreateLogEvent(logType string, x float64, y float64, node *utils.PeerNode) {
	client, err := utils.DbConnect()
	for client == nil || err != nil {
		client, err = utils.DbConnect()
	}
	defer client.Client().Disconnect(context.Background())
	workspaceId, _ := primitive.ObjectIDFromHex(*node.WorkspaceId)

	logEvent := utils.LogEvent{
		LogType:           logType,
		WorkspaceID:       workspaceId,
		BlockchainAddress: *node.BlockchainAddress,
		XDataPoint:        x,
		YDataPoint:        y,
		CreatedAt:         primitive.NewDateTimeFromTime(time.Now()),
	}

	logEventsCollection := client.Collection("logevents")
	_, err = logEventsCollection.InsertOne(context.Background(), logEvent)

	if err != nil {
		log.Fatal(err)
	}

}
